lib/tasks/test_unit.rake

```diff
@@ -0,0 +1,57 @@
+namespace :test do
+
+  # no seperation for tests by file but separation between unit and functional
+  Rake::TestTask.new do |t|
+    t.name = "all"
+    t.description = "Autotask - run unit and functional tests"
+    t.libs << "test"
+    t.pattern = FileList["test/unit/*_test.rb", "test/functional/*_test.rb"]
+    t.warning = false
+    t.verbose = true
+  end
+
+  # autogenerates a rake task for every test file added to test/unit
+  COMPUTE_UNIT_TASKS = []
+    Dir.glob("test/unit/*_test.rb").each do |task|
+    collection = task.gsub(/test\/unit\//, "")
+    Rake::TestTask.new(:"#{collection}") do |t|
+      t.libs << "test"
+      t.description = "Autotask - run unit tests - #{collection}"
+      t.pattern = FileList["test/unit/#{collection}"]
+      t.warning = false
+      t.verbose = true
+    end
+    COMPUTE_UNIT_TASKS << "#{collection}"
+  end
+
+  # runs unit tests, one file at a time, in file order
```

Comment:
If it's possible to annotate what actual CLI command is required for each of these, that'd be helpful! Thanks!

Reply:
@jywarren Yes will do! I added your notes from the other PR under "pending" in the description -- these will be addressed too! I'll add checkboxes for them as well so you can quickly see their status.

Reply:
@jywarren ok comments and integration tests added. Please let me know if you think they are clear! I am going to hold on Travis integration until @siaw23 advises further.